wallet/wallet-address-generator-lib/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::{Parser, ValueEnum};
+use common::address::pubkeyhash::PublicKeyHash;
+use common::address::Address;
+use common::chain::config::{Builder, ChainType};
+use common::chain::{ChainConfig, Destination};
+use crypto::key::extended::{ExtendedPrivateKey, ExtendedPublicKey};
+use crypto::key::hdkd::u31::U31;
+use crypto::key::hdkd::{child_number::ChildNumber, derivable::Derivable};
+use utils::ensure;
+use wallet::WalletError;
+use wallet::{
+    key_chain::{make_account_path, KeyChainError, MasterKeyChain},
+    WalletResult,
+};
+use wallet_types::KeyPurpose;
+
+#[derive(Clone, Debug, ValueEnum)]
+pub enum Network {
+    Mainnet,
+    Testnet,
+    Regtest,
+    Signet,
+}
+
+impl From<Network> for ChainType {
+    fn from(value: Network) -> Self {
+        match value {
+            Network::Mainnet => ChainType::Mainnet,
+            Network::Testnet => ChainType::Testnet,
+            Network::Regtest => ChainType::Regtest,
+            Network::Signet => ChainType::Signet,
+        }
+    }
+}
+
+#[derive(Parser, Debug)]
+#[clap(version)]
+pub struct CliArgs {
+    /// Which network to generate the addresses for
+    #[arg(long, value_enum, default_value_t = Network::Mainnet)]
+    pub network: Network,
+
+    /// Number of addresses to generate and display
+    #[clap(long, default_value_t = 20)]
+    pub number_addresses: u8,
+
+    /// Mnemonic phrase (12, 15, or 24 words as a single quoted argument). If not specified, a new mnemonic phrase is generated and printed.
+    #[clap(long)]
+    pub mnemonic: Option<String>,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum CliError {
+    #[error("Invalid input: {0}")]
+    InvalidInput(String),
+    #[error("Invalid mnemonic: {0}")]
+    InvalidMnemonic(wallet_controller::mnemonic::Error),
+    #[error("WalletError error: {0}")]
+    WalletError(#[from] WalletError),
+}
+
+pub fn run(args: CliArgs) -> Result<(), CliError> {
+    ensure!(
+        args.number_addresses <= 20,
```

Comment:
This number should come from the default gap in the wallet, I'd say. Is it possible to do that?

Reply:
Fixed used the LOOKAHEAD_SIZE constant from the keychain